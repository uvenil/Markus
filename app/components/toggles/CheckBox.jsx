'use strict';

import React from 'react';
import { observer } from 'mobx-react';
import ToggleStore from './ToggleStore';
import Unique from '../../utils/Unique';

@observer
export default class CheckBox extends React.Component {
    constructor(props) {
        super(props);

        this._checkBoxId = Unique.elementId();

        this._handleChange = checked => {
            if (this.props.onChange) {
                this.props.onChange(checked);
            }
        };
    }

    render() {
        const theme     = this.props.theme === 'dark' ? require('../../theme.dark.json') : require('../../theme.light.json');
        const textColor = this.props.textColor ? this.props.textColor : theme.primaryTextColor;

        return (
            <div
                className={this.props.className}
                style={{ verticalAlign : 'middle' }}>
                <input
                    id={this._checkBoxId}
                    type="checkbox"
                    checked={this.props.store.checked}
                    onChange={event => this._handleChange(event.target.checked)} />
                <label
                    htmlFor={this._checkBoxId}
                    style={{ WebkitUserSelect : 'none', cursor : 'default', fontFamily : this.props.fontFamily, fontSize : this.props.textSize, color : textColor }}>
                    {this.props.children}
                </label>
            </div>
        );
    }
}

CheckBox.propTypes = {
    store      : React.PropTypes.instanceOf(ToggleStore),
    className  : React.PropTypes.string,
    fontFamily : React.PropTypes.string,
    textSize   : React.PropTypes.string,
    textColor  : React.PropTypes.string,
    theme      : React.PropTypes.oneOf([ 'light', 'dark' ]),
    onChange   : React.PropTypes.func
};

CheckBox.defaultProps = {
    className : 'CheckBox',
    theme     : 'light'
};

module.exports = CheckBox;
