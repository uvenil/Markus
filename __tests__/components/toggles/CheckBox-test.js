'use strict';

import React from 'react';
import Renderer from 'react-test-renderer';
import CheckBox from '../../../app/components/toggles/CheckBox.jsx';

jest.mock('../../../app/utils/Unique');

it('renders correctly', () => {
    const checkBox = Renderer.create(
        <CheckBox>Hello, World!</CheckBox>
    ).toJSON();

    expect(checkBox).toMatchSnapshot();
});
