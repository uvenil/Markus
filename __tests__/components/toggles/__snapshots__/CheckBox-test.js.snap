exports[`test renders correctly 1`] = `
<div
  className="CheckBox"
  style={
    Object {
      "verticalAlign": "middle"
    }
  }>
  <input
    id="a"
    type="checkbox" />
  <label
    htmlFor="a"
    style={
      Object {
        "WebkitUserSelect": "none",
        "color": "#000",
        "cursor": "default",
        "fontFamily": undefined,
        "fontSize": undefined
      }
    }>
    Hello, World!
  </label>
</div>
`;
